### PR TITLE
API key control setting

### PR DESCRIPTION
**Pull request by zachmullen**
This is a new system setting that can be used to disable API key functionality on an instance. This is a security requirement for some of our users.

I'd like to disable the UI at some point too, but that will be easier if we can have settings with a public flag, which is likely a 3.0 change.

### CODE REVIEW

**Comment by manthey** [comment]
This disable the restful endpoint for generating api keys.  Do we need to disable them in the model layer, too?

**Comment by zachmullen** [comment]
> Do we need to disable them in the model layer, too?

Don't think so, this is just to disallow end users to use API keys.

**Comment by brianhelba** [comment]
@zachmullen Can you elaborate on how disabling API keys improves security?

**Comment by zachmullen** [comment]
> Can you elaborate on how disabling API keys improves security?

This will be used on an instance where users are only allowed to login via a specific OpenID provider that enforces a two-factor authentication policy. The sysadmins for this instance would not approve of API-key style authentication since it does not use two-factor authentication.

File: girder/api/v1/api_key.py [hunk 1]
```diff
@@ -18,10 +18,11 @@
 ###############################################################################
 
 from ..describe import Description, autoDescribeRoute
-from ..rest import Resource, filtermodel
+from ..rest import Resource, RestException, filtermodel
```
Comment (manthey) [inline girder/api/v1/api_key.py]: As of PR #2498, we're importing exceptions from `girder.exceptions`.

**Pull request merged**